File: TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        vector<int> res;
        for(int x = 0; x < nums.size(); x++){
            if(map.find(target-nums[x]) != map.end() && map[target-nums[x]] != x){
                res.push_back(map[target-nums[x]]);
                res.push_back(x);
                break;
            }
            map[nums[x]] = x;
        }
        return res;
    }
};

File: TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> d = new Dictionary<int,int>();
        int[] res = new int[2];
        for(int i = 0; i < nums.Length; i++){
            int value = -1;
            if(d.TryGetValue(target - nums[i], out value)){
                if(value != i){
                    res[0] = d[target - nums[i]];
                    res[1] = i;
                }
            }
            else{
                d[nums[i]] = i;
            }
        }
        return res;
    }
}
